### ISSUE TITLE
Git Branching Strategy

### ISSUE THREAD

**Issue by Southclaws** [pawn-lang/compiler#284]
So I've added some .github files and you'll notice some nice new callouts for issue and pull request creation:

![image](https://user-images.githubusercontent.com/1636971/35977683-aefba628-0cdb-11e8-9a9f-0d5082425438.png)

One of the clauses I've added which is common in Git workflows is to match `master` with the latest release at all times and keeping a `dev` branch to stage changes in.

The issue with introducing this now is we either have to rebase master on an earlier commit (revert) to make it match the latest release (3.10.4 is latest stable) or just wait for a new release and then adhere to this rule from now on.

cc @pawn-lang/compiler team, I'd like all your input on this 🙂

**Comment by Jeroen52** [comment]
Rebasing on a repo like this one should not be done, it requires everyone to set their revisions to the specific commit. And if they don't do that it'll cause an error.

But I do encourage you to use `dev` and `master`, even though I recently moved to only `master` in my development process on private projects to auto-deploy code as fast as possible without having to create 35+ merges.

**Comment by YashasSamaga** [comment]
In my opinion, the template should be small and concise. It shouldn't annoy the reporter with too much information. The writing spaces must not be broken down into too many pieces.

```
<!-- pawn-lang compiler issue template -->

<!--
Please ensure that your issue is directly related to the this pawn-lang/compiler. Issues
seeking help with scripting or doubts about SA-MP are not relavent to this project.
-->

### Issue type:

* [ ] Bug Report
* [ ] Feature Request
* [ ] Question
* [ ] Other

### Issue description:
<!--
Information that could be useful:
- how to reproduce the problem
- observed behaviour
- expected behaviour
- other relavent information
-->

### Minimal complete verifiable example (MCVC):
<!--
If relavent, please provide a minimal complete verifiable code which compiles
and reproduces the bug. If the compiler is showing undefined (random) behaviour,
provide code which invokes the undefined behaviour.

Learn more about writing MCVE from [StackOverflow](https://stackoverflow.com/help/mcve).
-->

` ``
` ``

<!-- comments on the code, if any -->

### Workspace Information:

* Compiler version:
* Operating System:
* Command line arguments provided:
* sampctl version (if applicable):

```
---





### Issue type:

* [ ] Bug Report
* [ ] Feature Request
* [ ] Question
* [ ] Other

### Issue description:


### Minimal complete verifiable code:


```
```



### Workspace Information:

* Compiler version:
* Operating System:
* Command line arguments provided:
* sampctl version (if applicable):

**Comment by Dayvison** [comment]
I agree with Yashas, and i don't see where sampctl fits on a pawn issue

**Comment by YashasSamaga** [comment]
My proposal for PR template:

```
<!-- pawn-lang pull-request template -->

<!--
Please read the contributing guidelines (CONTRIBUTING.md) before submitting pull-requests.

A pull request must address a single issue or a set of related issues. PRs addressing a wide range of unrelated issues will be closed without any consideration. Trivial pull-requests (for example, removing a space) will be closed immediately.
-->
<!-- Please ensure that the following your pull-request passes the following checklist: -->
#### Pre-submission checklist:

* [ ] PR targets `dev` branch
* [ ] Tested locally

#### Type of change(s):

* [ ] Bug fix
* [ ] New feature
* [ ] Meta

#### Summary:
* [ ] Adds new global symbols
* [ ] Breaking changes
* [ ] CI Tests present

#### Issue(s) addressed:
<!--
- one line per issue addressed
- issue must be tagged (#<issue no>)

Example:
Fixes #<issue no>
-->

#### Description:
<!---
Explain briefly the changes the pull-request makes.
-->
```





#### Pre-submission checklist:

* [ ] PR targets `dev` branch
* [ ] Tested locally

#### Type of change(s):

* [ ] Bug fix
* [ ] New feature
* [ ] Meta

#### Summary:
* [ ] Adds new global symbols
* [ ] Breaking changes
* [ ] CI Tests present

#### Issue(s) addressed:


#### Description:

**Comment by Zeex** [comment]
Nice, but why do we need a stable branch when we have release tags?

**Comment by Y-Less** [comment]
> why do we need a stable/release branch when we have release tags?

> Rebasing on a repo like this one should not be done
> i don't see where sampctl fits on a pawn issue

> the template should be small and concise. It shouldn't annoy the reporter with too much information.

I agree with all those points.

@YashasSamaga Asked for comments and votes.  There doesn't seem to be much dissenting opinion to vote about, so I just commented.

**Comment by Southclaws** [comment]
> i don't see where sampctl fits on a pawn issue

sampctl isn't a compiler, it actually uses *this* compiler with a set of specific flags and inputs. If a user is having issues with the compiler, it would be useful to know *how* they are invoking the compiler, be it an editor script, the command line or a wrapper program such as sampctl. It's one mention in one bullet point, I honestly don't see what's to gain from *not* asking the user how they invoked the compiler.

But anyway, I just did it quickly when I had time so use PRs/reviews to submit and discuss changes.

**Comment by YashasSamaga** [comment]
@Southclaws How does the editor matter if they are providing the command line arguments used (whatever the editor uses)?

**Comment by Southclaws** [comment]
Because not everyone understands what command line arguments are. If someone has simply downloaded a template (Sublime plugin generates a config, vscode has a template tasks.json).

Instead of asking a user "what arguments did you use" and then having to explain what arguments are, where to find the config file (differs based on system, absolute nightmare to track down sometimes) simply asking a user if they use Sublime with the pawn-sublime-language extension we can instantly infer the configuration from the defaults. If a user doesn't know what arguments are, it's unlikely they have changed from the defaults.

*It's better to have information that you don't need than need it when you don't have it.*

**Comment by YashasSamaga** [comment]
## New proposal for issue template

```
<!-- pawn-lang compiler issue template -->

<!--
Please ensure that your issue is directly related to the pawn-lang/compiler. Issues
seeking help with scripting or doubts about SA-MP are not relevant to this project.
-->

### Issue description:
<!--
Information that could be useful:
- how to reproduce the problem
- observed behaviour
- expected behaviour
- other relevant information
-->

### Minimal complete verifiable example (MCVE):
<!--
Please provide a minimal complete verifiable code which compiles
and reproduces the problem. If the compiler is showing undefined (random) behaviour,
provide code which invokes the undefined behaviour.

Learn more about writing MCVE from [StackOverflow](https://stackoverflow.com/help/mcve).

If this section is not relevant, feel free to remove this section from your issue.
-->

` ``
` ``

<!-- comments on the code, if any -->

### Workspace Information:

* Compiler version:
* Command line arguments provided (or sampctl version):
* Operating System:
```

---





### Issue description:


### Minimal complete verifiable example (MCVE):


` ``
` ``



### Workspace Information:

* Compiler version:
* Command line arguments provided (or sampctl version):
* Operating System:

---

---

### Issue type is redundant

The team members add labels to the issues which indicate what kind of issue it is. The checkpoints in the issue are just bloat and do not offer any additional information.

### sampctl version = command line arguments

Let both of them use the same row in the workspace information.

### explicitly allow issue makers to remove the MCVE off

When providing code is not relevant, the section becomes useless. Many issues that exist today leave unused sections as they are instead of removing them.

**Comment by Southclaws** [comment]
oops I forgot this issue existed, please see #333
